src/backend/utils/cache/plancache.c

```diff
@@ -542,18 +545,18 @@ RevalidateCachedQuery(CachedPlanSource *plansource, IntoClause *intoClause)
 	 */
 	if (intoClause)
 	{
-		CreateTableAsStmt *ctas_stmt;
+		SelectStmt *select;
 
-		if (!IsA(plansource->raw_parse_tree, CreateTableAsStmt))
+		if (!IsA(plansource->raw_parse_tree, SelectStmt))
 			ereport(ERROR,
 					(errcode(ERRCODE_WRONG_OBJECT_TYPE),
-					 errmsg("prepared statement is not a CreateTableAs Statement.")));
+					 errmsg("prepared statement is not a SELECT")));
 
-		ctas_stmt = (CreateTableAsStmt *) copyObject(plansource->raw_parse_tree);
+		select = (SelectStmt *) copyObject(plansource->raw_parse_tree);
 
-		ctas_stmt->into = copyObject(intoClause);
+		select->isCtas = true;
```

Comment:
Maybe here we should set intoClause and isCtas is not needed for SelectStmt. Since for CTAS we are replanning, and we are replanning using the 'SELECT INTO' syntax.

Reply:
If `into` is set for the statement, the parser and analyzer would treat it as a `CreateTableAsStmt`, not a `SelectStmt` anymore. See function `transformTopLevelStmt()`.
And the result of the revalidation would be an UtilityStmt, not a PlannedStmt. The is not the expected result, and will error out in `ExecuteQuery()`.